src/actions/chat_actions.js

```diff
@@ -75,77 +65,63 @@ class ChatActions {
     });
   }
 
-  init(url, token) {
-    this.interval = setInterval(getMessages(url, token), 500);
-  }
-
-  joinChannel(url, token) {
-    $.ajax({
-      url: url,
-      method: 'PUT',
-      dataType: 'json',
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'Content-Type': 'application/json'
-      },
-      success(membershipObj) {
+  joinChannel(room) {
+    SocketStore.getSocket().emit("join", room, (response) => {
+      if(response.Success){
         Dispatcher.dispatch({
           actionType: ActionTypes.MEMBERSHIP_RECEIVED,
-          membership: membershipObj.membership.room_id
+          membership: response.Result.room_id
         });
-      },
-      error() {
-        console.log(arguments);
+      }else{
+        console.log(response);
       }
-    });
+    })
   }
 
-  leaveChannel(url, token) {
-    $.ajax({
-      url: url,
-      method: 'DELETE',
-      dataType: 'json',
-      headers: {
-        Authorization: `Bearer ${token}`,
-        'Content-Type': 'application/json'
-      },
-      success(membershipObj) {
+  leaveChannel(room) {
+    SocketStore.getSocket().emit("leave", room, (response) => {
+      if(response.Success){
         Dispatcher.dispatch({
           actionType: ActionTypes.MEMBERSHIP_DESTROYED,
-          membership: membershipObj.deleted
+          membership: response.Result
         });
-      },
-      error() {
-        console.log(arguments);
+      }else{
+        console.log(response);
       }
+    })
+  }
+
+  submitMessage(room, body) {
+    SocketStore.getSocket().emit("message", {Room: room, Body:body}, function(response){
+        if(response.Success){
+            Dispatcher.dispatch({
+                actionType: ActionTypes.CHAT_MESSAGE_RECEIVED
+            });
+        }
+    });
+    Dispatcher.dispatch({
+      actionType: ActionTypes.CHAT_MESSAGE_SUBMITTED,
+      message: body
     });
   }
 
-  submitMessage(url, token, message) {
+  getMessages(channel){
+    const url = AppStore.getUrl() + `/rooms/${channel}/messages`;
     $.ajax({
       url: url,
-      method: 'POST',
       dataType: 'json',
-      data: JSON.stringify({
-        body: message.get('body')
-      }),
       headers: {
-        Authorization: `Bearer ${token}`,
-        'Content-Type': 'application/json'
+        Authorization: `Bearer ${CurrentUserStore.getToken()}`
```

Comment:
Okay, cool, thanks! Will be curious to hear what the latest is.
On Thu, Mar 12, 2015 at 7:50 PM Frederik Creemers <notifications@github.com>
wrote:

> In src/actions/chat_actions.js
> <https://github.com/asm-products/landline-web/pull/4#discussion_r26361874>
> :
>
> >        headers: {
> > -        Authorization: `Bearer ${token}`,
> > -        'Content-Type': 'application/json'
> > +        Authorization: `Bearer ${CurrentUserStore.getToken()}`
>
> I think that actions accessing data directly from stores doesn't break the
> one-directionality of data. In the original flux schema, it looks like this.
>
> View -> Action -> Dispatcher -> Store-|
> ^----------^-------------------------------------<
>
> I'm hopping in to the #reactjs channel on freenode to ask whether people
> know of any drawbacks to this. All updates to the data still go through the
> dispatcher, so the changes to data are still atomic and easy to reason
> about.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asm-products/landline-web/pull/4/files#r26361874>.
>